### ISSUE TITLE
failure to initialize on centos6

### ISSUE THREAD

**Issue by p3ck** [CentOS-PaaS-SIG/linchpin#388]
using pinfile and topology from https://github.com/herlo/lp_test_workspace

```
linchpin up libvirt
/usr/lib64/python2.6/site-packages/cryptography/__init__.py:26: DeprecationWarning: Python 2.6 is no longer supported by the Python core team, please upgrade your Python. The next version of cryptography will drop support for Python 2.6
  DeprecationWarning
/usr/lib/python2.6/site-packages/linchpin/api/context.py:89: DeprecationWarning: The SafeConfigParser class has been renamed to ConfigParser in Python 3.2. This alias will be removed in future versions. Use ConfigParser directly instead.
  config = ConfigParser.SafeConfigParser()
/usr/lib/python2.6/site-packages/linchpin/api/context.py:91: DeprecationWarning: This method will be removed in future versions.  Use 'parser.read_file()' instead.
  config.readfp(f)
Traceback (most recent call last):
  File "/usr/bin/linchpin", line 11, in <module>
    load_entry_point('linchpin==1.0.4', 'console_scripts', 'linchpin')()
  File "/usr/lib/python2.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python2.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python2.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python2.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python2.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python2.6/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/usr/lib/python2.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python2.6/site-packages/linchpin/__init__.py", line 176, in up
    return_code, results = lpcli.lp_up(pf_w_path, targets, run_id=run_id)
  File "/usr/lib/python2.6/site-packages/linchpin/api/__init__.py", line 245, in lp_up
    return self.run_playbook(pinfile, targets, action="up", run_id=run_id)
  File "/usr/lib/python2.6/site-packages/linchpin/api/__init__.py", line 528, in run_playbook
    self.ctx.log_debug('rundb_id: {}'.format(rundb_id))
ValueError: zero length field name in format
```

**Comment by herlo** [comment]
@p3ck is this using anything special? Are you using a virtualenv? How did you install? I'm looking to duplicate this, but have been installing/running this on CentOS without issue for some time.

**Comment by p3ck** [comment]
@herlo just to follow up here I believe you said some code was introduced that doesn't work under python2.6 which is what centos6 is.

Now that you have the Dockerfiles I used do you need anything else from me?

**Comment by herlo** [comment]
@p3ck I figured out this specific issue, but there are further issues with centos 6 that I have identified. I just merged #390 so I can use it in the mainline code for testing.

I'm going to leave this ticket open and address these as I can. I suspect linchpin only works on certain platforms as I couldn't get it to work on centos 6 or fedora 25. I'll work through your setup and give feedback as I can. Nothing else is needed at this time. Thanks!

**Comment by herlo** [comment]
@p3ck, actually. I wonder if you could file a bug for the latest errors on each distro from your testing. I'm wanting to identify those issues and think that would be supper helpful.

**Comment by arilivigni** [comment]
@herlo fedora 25 should 100% be supported.  I am not as concerned about CentOS6.

**Comment by herlo** [comment]
@arilivigni I agree, but it is currently failing as well. @p3ck chose the four distros to test, so I went with what he had.

**Comment by p3ck** [comment]
@herlo what distros do you currently test against when you make a release?

Currently the issues I have hit are as follows:
#387 - fails on centos7, Fedora25 and Fedora26.
#388 - this issue which fails on Centos6.  I imagine if this is fixed it will also fail on #387.

I have not had an issue installing linchpin on Fedora25 using the Docker images.  I don't have anything more to report until these issues are fixed.

**Comment by herlo** [comment]
@p3ck the distro I test against is F26 in a virtualenv. I test it on centos7 as I can as well. Fedora 25 should be tested, but isn't. I think CentOS 6 is a useful test, but as pointed out above is less important. 

@arilivigni what do you think of F27? I've not yet upgraded, but figure it's going to need to be in the pipe soon. Yes?

**Comment by herlo** [comment]
Fixed by #398
